src/init.c

```diff
@@ -210,13 +210,19 @@ JL_DLLEXPORT void jl_atexit_hook(int exitcode)
     // upset by the local mess of things when we run the user's _atexit hooks
     jl_task_frame_noreturn(ct);
 
+    if (ct == NULL)
+        ct = container_of(jl_adopt_thread(), jl_task_t, gcstack);
```

Comment:
I initially only did this if `jl_base_module` is defined. Otherwise you might try to call this early in bootstrapping if something went wrong, and crash badly instead.

Reply:
Ok check again.